modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -584,6 +584,19 @@ Mat::Mat(const std::initializer_list<_Tp> list)
         return;
     Mat((int)list.size(), 1, traits::Type<_Tp>::value, (uchar*)list.begin()).copyTo(*this);
 }
+
+template<typename _Tp, typename> inline
+Mat::Mat(const std::initializer_list<int> sizes, const std::initializer_list<_Tp> list)
+    : Mat()
+{
+    size_t size_total = 1;
+    int *sz = (int*)sizes.begin();
+    for(int i = 0; i < sizes.size(); i++)
```

Comment:
```.diff
-int i = 0
+size_t i = 0
```

Because "sizes.size()" has size_t type (almost, http://en.cppreference.com/w/cpp/utility/initializer_list/size ).

Anyway it is better to use C++11 "foreach":

```
for (auto s : sizes)
{
    size_total *= s;
}
```

Reply:
>for (auto s : sizes)
>{
>    size_total *= s;
>}
->I think that this code is most simplified and suitable.